package/MDAnalysis/core/Selection.py

```diff
@@ -490,23 +559,22 @@ class StringSelection(Selection):
 
     Supports the use of wildcards at the end of strings
     """
-    badtokens = {'(', ')', 'and', 'or', 'not', 'segid', 'resid', 'resname'
-                 'name', 'type'}
-
     def __init__(self, parser, tokens):
-        data = tokens.popleft()
-        if data in self.badtokens:
-            raise ValueError("Unexpected token: {}".format(data))
+        vals = grab_not_keywords(tokens)
+        if not vals:
+            raise ValueError("Unexpected token '{0}'".format(tokens[0]))
 
-        self.val = data
+        self.values = vals
 
     def apply(self, group):
-        wc_pos = self.val.find('*')
-        if wc_pos == -1:  # No wildcard found
-            mask = getattr(group, self.field) == self.val
-        else:
-            values = getattr(group, self.field).astype(np.str_)
-            mask = np.char.startswith(values, self.val[:wc_pos])
+        mask = np.zeros(len(group), dtype=np.bool)
+        for val in self.values:
+            wc_pos = val.find('*')
+            if wc_pos == -1:  # No wildcard found
+                mask |= getattr(group, self.field) == val
```

Comment:
It could be applied to other selections maybe....

The other major pattern we have is `modifier <other selection>`, so bonds works like `bonded <other selection>`, so to implicitly allow OR with selections could end up with a string like:

`name H and bonded name O type N` == `name H and bonded (name O or type N)`

`not around 7.0 protein mass > 14.0` == `not (around 7.0 protein) or (mass > 14.0)`

But I'm not sure those are as readable because they're naturally longer in length.